### PR TITLE
refactor: add npm run format && npm run lint

### PR DESCRIPTION
**Pull request by tim775**
Adds our standard format and lint scripts, plus converts to typescript.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    'no-useless-constructor': 'off',
+  },
+  overrides: [
+    {
+      files: ['src/api/resolvers/**/*'],
+      rules: {
+        'import/prefer-default-export': 'off',
+      }
+    },
+  ],
```
Comment (aliscott) [inline .eslintrc.js]: We shouldn't need this here.

File: package.json [hunk 2]
```diff
@@ -4,10 +4,11 @@
   "description": "",
   "main": "dist/index.js",
   "scripts": {
-    "lint": "eslint .",
+    "lint": "eslint --ext .ts,.js --ignore-path .gitignore --ignore-path .eslintignore .",
     "prepare": "ncc build src/index.ts -o dist --source-map --license licenses.txt",
     "test": "jest",
-    "all": "npm run lint && npm run prepare && npm run test"
+    "all": "npm run lint && npm run prepare && npm run test",
+    "format": "prettier --write 'src/**/*.{js,ts}'"
```
Comment (aliscott) [inline package.json:11]: We should also copy the `.prettierrc` file into this repo

File: latest.json [hunk 3]
```diff
@@ -0,0 +1,189 @@
+{
```
Comment (aliscott) [inline latest.json]: What's this file for?

**Review by aliscott — APPROVED** [review]

File: latest.json (hunk 3 above)
Comment (tim775) [reply latest.json]: debugging cruft

**Pull request merged**